### PR TITLE
Improve BigInt support

### PR DESCRIPTION
**Pull request by zherczeg**
This patch adds several small features:
- Support right shift with negative numbers
- Support exponentation operator
- BigInts can be enabled in ES5.1 mode
- Remove dead code from ecma_deref_bigint
- Support longer BigInt literals in the parser
- Fix various BigInt comparison issues
- Do not discard unary plus for BigInt constants

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-bigint.c [hunk 1]
```diff
@@ -1270,10 +1296,102 @@ ecma_bigint_shift (ecma_value_t left_value, /**< left BigInt value */
...
+  {
+    result_p = ecma_big_uint_pow (left_p, power);
+  }
+
+  if (JERRY_UNLIKELY (result_p == NULL))
+  {
+    return ecma_bigint_raise_memory_error ();
+  }
+
+  if ((left_p->u.bigint_sign_and_size & ECMA_BIGINT_SIGN) && (power & 0x1) != 0)
```
Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-bigint.c:1385]: `power & ECMA_BIGINT_SIGN`

File: jerry-core/ecma/operations/ecma-comparison.c [hunk 2]
```diff
@@ -192,7 +192,10 @@ ecma_op_abstract_equality_compare (ecma_value_t x, /**< first operand */
       return ecma_make_boolean_value (ecma_bigint_is_equal_to_number (x, ecma_get_number_from_value (y)));
     }
 
-    return ECMA_VALUE_FALSE;
+    /* Swap values. */
+    ecma_value_t tmp = x;
+    x = y;
+    y = tmp;
```
Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-comparison.c:198]: You can do this without allocating a third temporary variable:
```
x ^= y;
y ^= x;
x ^= y;
```

File: jerry-core/ecma/operations/ecma-big-uint.c [hunk 3]
```diff
@@ -94,23 +94,39 @@ ecma_big_uint_extend (ecma_extended_primitive_t *value_p, /**< BigUInt value */
 } /* ecma_big_uint_extend */
 
 /**
- * Discard the high-order digits of a BigUInt
+ * Helper function which discards the leading zero digits of a BigUInt value
  *
- * @return BigUInt value, NULL on error
+ * @return ecma BigUInt value or ECMA_VALUE_ERROR
+ *         Returned value must be freed with ecma_free_value.
```
Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-big-uint.c:100]: > Returned value must be freed with ecma_free_value.

Is this correct? The return value is not `ecma_value_t`

File: jerry-core/ecma/operations/ecma-big-uint.c [hunk 4]
```diff
@@ -1410,71 +1488,91 @@ ecma_big_uint_shift_right (ecma_extended_primitive_t *left_value_p,  /**< left B
...
+      ecma_deref_bigint (square_p);
+    }
 
-  JERRY_ASSERT (size > 0);
+    square_p = new_square_p;
+    right_value >>= 1;
 
-  ecma_bigint_digit_t *first_digit_p = ECMA_BIGINT_GET_DIGITS (value_p, 0);
-  ecma_bigint_digit_t *last_digit_p = ECMA_BIGINT_GET_DIGITS (value_p, size);
+    if ((right_value & 0x1) != 0)
```
Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-big-uint.c:1536]: Magic number literal, `(right_value & ECMA_BIGINT_SIGN)`

File: jerry-core/ecma/operations/ecma-bigint.c (hunk 1 above)
Comment (zherczeg) [reply jerry-core/ecma/operations/ecma-bigint.c:1385]: It is an even/odd testing, it is not related to ECMA_BIGINT_SIGN. Shall I add a comment?

File: jerry-core/ecma/operations/ecma-comparison.c (hunk 2 above)
Comment (zherczeg) [reply jerry-core/ecma/operations/ecma-comparison.c:198]: This is copied from above. Shall I change all of them or keep this method for now?

File: jerry-core/ecma/operations/ecma-big-uint.c (hunk 4 above)
Comment (zherczeg) [reply jerry-core/ecma/operations/ecma-big-uint.c:1536]: This is also not related to sign, it is checked whether the last bit is 1. I can add a macro.

File: jerry-core/ecma/operations/ecma-comparison.c (hunk 2 above)
Comment (zherczeg) [reply jerry-core/ecma/operations/ecma-comparison.c]: Rewritten.

**Pull request merged**